controllers/webhooks/networking/security_groups/validator.go

```diff
@@ -0,0 +1,78 @@
+package security_groups
+
+import (
+	"context"
+	"fmt"
+
+	korifiv1alpha1 "code.cloudfoundry.org/korifi/controllers/api/v1alpha1"
+	"code.cloudfoundry.org/korifi/controllers/webhooks"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const SecurityGroupEnyityType = "securityGroup"
+
+var cfsecuritygrouplog = logf.Log.WithName("cfsecuritygroup-validation")
+
+//+kubebuilder:webhook:path=/validate-korifi-cloudfoundry-org-v1alpha1-cfsecuritygroup,mutating=false,failurePolicy=fail,sideEffects=None,groups=korifi.cloudfoundry.org,resources=cfsecuritygroups,verbs=create;update;delete,versions=v1alpha1,name=vcfsecuritygroup.korifi.cloudfoundry.org,admissionReviewVersions={v1,v1beta1}
+
+type Validator struct {
+	duplicateValidator webhooks.NameValidator
+	rootNamespace      string
+}
+
+var _ webhook.CustomValidator = &Validator{}
+
+func NewValidator(duplicateValidator webhooks.NameValidator, rootNamespace string) *Validator {
```

Comment:
We should implement destination and port [validation](https://github.com/cloudfoundry/korifi/pull/3858/files#diff-3ee3f212d5c30f2623d51977a5a65a43e557442d47abd133326bb5fbf7b7424dR59) here as well (on create and update) in order to guarantee that a kubectl user is not bypassing API request validation

Then if we had a webhook that validates create and update, then the API could simply propagate the webhook error to the user, i.e. it does not need to implement the very same checks in the payload validation

We should look into how such errors look form the cf cli point of view though, i.e. are they reasonably presented to the user